### ISSUE TITLE
The preview function shows mojibake in gui, maybe wrong encoding? Where to edit the source code for preview? steve8000818@gmail.com

### ISSUE THREAD

**Issue by nissansz** [buliasz/tesstrain-windows-gui#13]
The preview function shows mojibake in gui, maybe wrong encoding? Where to edit the source code for preview? steve8000818@gmail.com

![image](https://user-images.githubusercontent.com/41010669/174195199-f295db3f-2fa0-4756-bbc5-a16003e345dc.png)

**Comment by buliasz** [comment]
Preview code is in the `_recognition_preview.ahki` file. But I think the function that probably needs to be fixed is `OcrImageFile` which is in the `_helper.ahki` file. This one leads to the `ExecuteCommand` function from the `_console.ahki`.
I guess the issue is that StdOut from cmd.exe is in your local windows code page.

**Comment by buliasz** [comment]
You can try fixing it by yourself or you can send me some example Ground Truth data and I'll do it when I find some time for it.

**Comment by buliasz** [comment]
Look 2 messages above your question. Already answered.

**Comment by nissansz** [comment]
[100split.zip](https://github.com/buliasz/tesstrain-windows-gui/files/8923496/100split.zip)
Check this sample ground truth data

**Comment by nissansz** [comment]
[jp100.zip](https://github.com/buliasz/tesstrain-windows-gui/files/8923562/jp100.zip)
check trainneddata here

**Comment by nissansz** [comment]
I tried to update with your source code, it does not work. Same wrong display result.

**Comment by buliasz** [comment]
I've fixed it. Please try the new version.

**Comment by nissansz** [comment]
Thank you.
But there is error when I tried to preview.
![image](https://user-images.githubusercontent.com/41010669/174210282-28f07eec-5f1a-4f9b-a76b-5da22e4d7ef8.png)

**Comment by buliasz** [comment]
I don't understand Japanese, what does the exception message say?

**Comment by nissansz** [comment]
The system cannot find the specified file.

**Comment by nissansz** [comment]
which file did you updated? Maybe I can copy your new updated file to old version to make it work.

**Comment by buliasz** [comment]
Fixed, check it now

**Comment by nissansz** [comment]
thank you.
Will check later.
Also, for training datasets, can I put them in several sub folders for each font?
Did it support sub folders?

**Comment by buliasz** [comment]
I don't think so. At the moment it executes commands the same way console training script does (from the main tesstrain repo). 
In the "Disscusions" section there's an "Ideas" category. You can use it to propose that. Maybe some day I'll have time to implement it or you or someone else will try doing it.
You can use just naming convention if the only think you want to achieve is have the input data ordered.

**Comment by buliasz** [comment]
You can use separate folders for different Ground Truth input and to train different .traineddata targets, if that's your point.

**Comment by nissansz** [comment]
I mean training for all fonts in a parent folder.
In paddle ocr, it supports reading label file with path for each image

**Comment by nissansz** [comment]
I mean training for all fonts in a parent folder.
In paddle ocr, it supports reading label file with path for each image

**Comment by nissansz** [comment]
![image](https://user-images.githubusercontent.com/41010669/174217257-3ca37f00-94e5-4907-ac72-69ca572a0cf7.png)

It can be displayed, but there is one problem.
The blank area is wrongly recognized as English letters.

**Comment by buliasz** [comment]
I think this is just a matter of more training.

**Comment by nissansz** [comment]
Ok. It can be solved by opencv cropping, but don't know how to execute in your source code.

**Comment by buliasz** [comment]
I assume the issue from subject is now fixed.
